core/src/main/java/org/elasticsearch/Build.java

```diff
@@ -88,10 +89,13 @@
     private final boolean isSnapshot;
 
     /**
-     * Returns path to elasticsearch codebase path
+     * The location of the code source for Elasticsearch
+     *
+     * @return the location of the code source for Elasticsearch which may be null
      */
     static URL getElasticsearchCodebase() {
```

Comment:
Maybe just change this to return String so the null handling is completely inside here?

Reply:
This was a good suggestion to pursue but we still need the `URL` instance so we will keep it like this.